frame/staking/src/lib.rs

```diff
@@ -286,9 +283,83 @@ pub mod weights;
 pub use weights::WeightInfo;
 
 pub mod inflation;
-pub mod offchain_election;
 pub mod slashing;
 
+pub mod migrations {
+	use super::*;
+
+	pub mod v6 {
+		use super::*;
+		use frame_support::{pallet_prelude::*, traits::Get, weights::Weight};
+
+		macro_rules! generate_storage_types {
+			($name:ident => Value<$value:ty>) => {
+				paste::paste! {
+					struct [<$name Instance>];
+					impl frame_support::traits::StorageInstance for [<$name Instance>] {
+						fn pallet_prefix() -> &'static str {
+							"DarwiniaStaking"
+						}
+						const STORAGE_PREFIX: &'static str = stringify!($name);
+					}
+					type $name = StorageValue<[<$name Instance>], $value, ValueQuery>;
+				}
+			};
+		}
+
+		// NOTE: value type doesn't matter, we just set it to () here.
+		generate_storage_types!(SnapshotValidators => Value<()>);
+		generate_storage_types!(SnapshotNominators => Value<()>);
+		generate_storage_types!(QueuedElected => Value<()>);
+		generate_storage_types!(QueuedScore => Value<()>);
+		generate_storage_types!(EraElectionStatus => Value<()>);
+		generate_storage_types!(IsCurrentSessionFinal => Value<()>);
+
+		/// check to execute prior to migration.
+		pub fn pre_migrate<T: Config>() -> Result<(), &'static str> {
+			// these may or may not exist.
+			log!(
+				info,
+				"SnapshotValidators.exits()? {:?}",
+				SnapshotValidators::exists()
+			);
+			log!(
+				info,
+				"SnapshotNominators.exits()? {:?}",
+				SnapshotNominators::exists()
+			);
+			log!(info, "QueuedElected.exits()? {:?}", QueuedElected::exists());
+			log!(info, "QueuedScore.exits()? {:?}", QueuedScore::exists());
+			// these must exist.
+			assert!(
+				IsCurrentSessionFinal::exists(),
+				"IsCurrentSessionFinal storage item not found!"
+			);
+			assert!(
+				EraElectionStatus::exists(),
+				"EraElectionStatus storage item not found!"
+			);
```

Comment:
What will happen if these `assert!` fail.

Reply:
This is a `fn pre_migrate`

And this only work when you use the `try-runtime` subcommand